datamol/io.py

```diff
@@ -544,3 +545,129 @@ def to_xlsx(
 
     with fsspec.open(urlpath, mode="wb") as f:
         PandasTools.SaveXlsxFromFrame(mols, f, molCol=mol_column, size=mol_size)
+
+
+EXTENSIONS_DICT = {
+    "csv": [
+        ".csv",
+        ".csv.gz",
+        ".csv.bz2",
+        ".csv.zip",
+        ".csv.xz",
+        ".csv.zst",
+        ".csv.tar",
+        ".csv.tar.gz",
+        ".csv.tar.xz",
+        ".csv.tar.bz2",
+    ],
+    "excel": [".xlsx"],
+    "parquet": [".parquet"],
+    "json": [
+        ".json",
+        ".json.gz",
+        ".json.bz2",
+        ".json.zip",
+        ".json.xz",
+        ".json.zst",
+        ".json.tar",
+        ".json.tar.gz",
+        ".json.tar.xz",
+        ".json.tar.bz2",
+    ],
+    "sdf": [
+        ".sdf",
+        ".sdf.gz",
+    ],
+}
+
+
+def _guess_filetype(path: str):
+    """Return a filetype given an input path. Filetypes returned can be from
+    `csv, excel, parquet, json, sdf`.
+    """
+    for name, extensions in EXTENSIONS_DICT.items():
+        for ext in extensions:
+            if path.endswith(ext):
+                return name
+
+
+# I was hoping to use something along the lines of this for checking kwargs
+# I dont think I'm correctly implementing it right now
+def _check_file_kwargs(
```

Comment:
Not a bad idea and intention here, but if one of them is not valid then the downstream functions will anyway raise an error (and probably with a better and insightful error message).

So, I feel it's most likely best to simply not use such a function.